model/patch_lifecycle.go

```diff
@@ -685,14 +667,18 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		)
 	}
 
+	// Newer patches always stored the parser project during patch creation.
```

Comment:
The logic is here is needed because patches that were created but weren't finalized yet before the deploy of EVG-18700 can't finalize without the old `PatchedParserProject` handling. We don't have a policy on things like how old patches can be before we disallow modifications, so I opted to allow modifying old patches.

I made a ticket to eventually remove the `PatchedParserProject` fallback (EVG-19636).